taca/analysis/analysis_nanopore.py

```diff
@@ -361,130 +361,132 @@ def process_minion_delivery_run(minion_run):
+                logger.info(
+                    f"Successfully updated the db entry of run {ont_run.run_id}"
+                )
 
-        # Transfer the MinKNOW .html report file to ngi-internal, renaming it to the full run ID. Requires password-free SSH access.
-        report_dest_path = os.path.join(
-            CONFIG["nanopore_analysis"]["ont_transfer"]["minknow_reports_dir"],
-            f"report_{ont_run.run_id}.html",
-        )
-        transfer_object = RsyncAgent(
-            glob_html[0],
-            dest_path=report_dest_path,
-            validate=False,
-        )
-        try:
-            transfer_object.transfer()
+            else:
+                logger.info(
+                    f"Run {ont_run.run_id} has not finished sequencing, do nothing."
+                )
+
+        # if the run document is marked as "finished"
+        if sesh.check_run_status(ont_run) == "finished":
             logger.info(
-                f"Successfully transferred the MinKNOW report of run {ont_run.run_id}"
+                f"Run {ont_run.run_id} exists in the database as an finished run."
             )
-        except RsyncError:
-            msg = f"An error occurred while attempting to transfer the report {glob_html[0]} to {report_dest_path}"
-            logger.error(msg)
-            raise RsyncError(msg)
 
+            glob_html = glob.glob(ont_run.run_dir + "/report*.html")
+            if len(glob_html) == 0:
+                error_message = f"Run {ont_run.run_id} is marked as finished, but missing .html report file."
+                logger.error(error_message)
+                raise AssertionError(error_message)
+            elif len(glob_html) > 1:
+                error_message = f"Run {ont_run.run_id} is marked as finished, but contains conflicting .html report files."
+                logger.error(error_message)
+                raise AssertionError(error_message)
 
-def transfer_ont_run(ont_run):
-    """Transfer ONT runs to HPC cluster."""
-    email_recipients = CONFIG.get("mail").get("recipients")
-    logger.info("Processing run {}".format(ont_run.run_id))
+            logger.info(f"Transferring the run report to ngi-internal.")
 
-    # Update StatusDB
-    try:
-        ont2couch(ont_run)
+            # Transfer the MinKNOW .html report file to ngi-internal, renaming it to the full run ID. Requires password-free SSH access.
+            report_dest_path = os.path.join(
+                CONFIG["nanopore_analysis"]["ont_transfer"]["minknow_reports_dir"],
+                f"report_{ont_run.run_id}.html",
+            )
+            transfer_object = RsyncAgent(
+                glob_html[0],
+                dest_path=report_dest_path,
+                validate=False,
+            )
+            try:
+                transfer_object.transfer()
+                logger.info(
+                    f"Successfully transferred the MinKNOW report of run {ont_run.run_id}"
+                )
+            except RsyncError:
```

Comment:
It might be good to include the original error message from the RsyncError here, to make troubleshooting easier.

Reply:
The RsyncError message should be caught by the `except Exception as e` on line 491 and added to the email message, so I think this is already the case?

Reply:
Ah, right, I missed that. Great! 👍